### PR TITLE
Updating Go, dependencies, and source images

### PR DESCRIPTION
**Pull request by cin**
> This has a good bit of work left. The `github.com/caarlos0/env/v6` dependency isn't being found for some reason in the code.  There are many changes to the controller-runtime package that need to be investigated closely. I was just trying to get things up to date with the API changes but I'm sure I broke some tests around the `MetaNew` and `MetaOld` objects as they don't exist anymore. I simply removed the references to try to get things to at least compile for now.

The issues mentioned above have been worked out. I had to disable calling goimport due to the issue above. I also had to explicitly include v0.4.0 of `github.com/go-logr/logr` bc zapr was referring to an older version of logr for some reason. Basically zapr couldn't find one of the new functions in  logr.

There is an opportunity to clean up the boilerplate in predicate_test.go but I'm interested to get feedback on it first.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,19 +16,19 @@ CRD_OPTIONS ?= "crd:crdVersions=v1,trivialVersions=false"
 all: test varnish-operator varnish-controller
 
 # Run tests
-test: generate fmt vet manifests
```
Comment (kscharm) [inline Makefile:19]: What is the reason for removing `fmt` from all of these commands?

File: pkg/varnishcluster/controller/varnishcluster_secret_test.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ import (
 
 var _ = Describe("the varnish secret", func() {
 	validBackendPort := intstr.FromInt(8080)
-	vcNamespace := "varnish-secret"
+	vcNamespace := "default" // TODO: how to create varnish-secret keyspace in mock?
```
Comment (kscharm) [inline pkg/varnishcluster/controller/varnishcluster_secret_test.go:23]: Can you provide more detail about what this TODO means?

**Review by kscharm — APPROVED** [review]
LGTM, I see mostly controller runtime changes and naming changes.

File: Makefile (hunk 1 above)
Comment (cin) [reply Makefile:19]: This is something we need to discuss actually. the makefile's fmt target is commented out for now. It was calling `goimports` which kept removing `github.com/caarlos0/env/v6` for some reason.

Comment (cin) [reply Makefile:19]: This is the error:
```
> make test
/Users/cin/go/bin/controller-gen object:headerFile=./hack/boilerplate.go.txt paths="./..."
cd /Users/cin/go/src/varnish-operator/ && goimports -w ./pkg ./cmd ./api
cd /Users/cin/go/src/varnish-operator/ && go vet ./pkg/... ./cmd/... ./api/...
# github.com/ibm/varnish-operator/pkg/varnishcluster/config
pkg/varnishcluster/config/config.go:47:34: undefined: env
pkg/varnishcluster/config/config.go:56:12: undefined: env
# github.com/ibm/varnish-operator/pkg/varnishcontroller/config
pkg/varnishcontroller/config/config.go:40:16: undefined: env
pkg/varnishcontroller/config/config.go:49:23: undefined: env
pkg/varnishcontroller/config/config.go:55:30: undefined: env
pkg/varnishcontroller/config/config.go:61:11: undefined: env
make: *** [vet] Error 2
```

Comment (cin) [reply Makefile]: old goimports version bit me here...

File: pkg/varnishcluster/controller/varnishcluster_secret_test.go (hunk 2 above)
Comment (cin) [reply pkg/varnishcluster/controller/varnishcluster_secret_test.go:23]: If the namespace doesn't exist, the test will fail now whereas it didn't matter with the older k8s client version. Wondering if we should look into making this work with the custom namespace name.

Comment (kscharm) [reply pkg/varnishcluster/controller/varnishcluster_secret_test.go:23]: Oh interesting, I don't think it's too difficult to created a mock namespace...

File: go.mod [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
-	k8s.io/apimachinery v0.17.4
-	k8s.io/client-go v0.17.4
-	sigs.k8s.io/controller-runtime v0.5.2
+	github.com/prometheus/common v0.17.0
+	go.uber.org/zap v1.16.0
+	k8s.io/api v0.20.4
+	k8s.io/apimachinery v0.20.4
+	k8s.io/client-go v0.20.4
+	sigs.k8s.io/controller-runtime v0.8.2
+	sigs.k8s.io/controller-tools v0.5.0 // indirect
```
Comment (tomashibm) [inline go.mod:24]: that line doesn't seem right. We shouldn't need the `controller-tools` to be imported.
could you run `go mod tidy` to see if it disappears? That command removes not imported libraries and adds imported. Should fix it if it get there by accident.

**Review by tomashibm — APPROVED** [review]
Looks good!
One minor thing that needs to be checked, but approving to not hold you back.

File: go.mod (hunk 3 above)
Comment (cin) [reply go.mod:24]: `go mod tidy` cleaned this one up. I'm not sure why, but it added another indirect import though. `k8s.io/apiextensions-apiserver v0.20.2`, which I bumped up to match all the other k8s imports (v0.20.4).

**Pull request merged**